### DISCUSSION TITLE
Iteration is reversed when used with x-if

### DISCUSSION THREAD

**Discussion by Jelenkee** [alpinejs/alpine#2209] [7. Bugs (archived)]
https://codepen.io/kuhschubser/pen/zYdYpOY

**Comment by Tim-Wils** [comment]
You could solve this in 2 ways: you can wrap your inner template with an element like so: < template >< div >< template >, but in your case I’d probably leave out the x-if template and put an x-show in the < li > element.

I personally hope for a built-in core solution for using templates with x-for and x-if or a solution for using <template> following a < template > 👍🏻
